Controller/NewsController.php

```diff
@@ -64,16 +64,37 @@ public function getNewsFieldsAction()
      *
      * @Get("news")
      *
-     * @return \Symfony\Component\HttpFoundation\Response
+     * @return Response
      */
     public function getNewsListAction(Request $request)
     {
         $restHelper = $this->get('sulu_core.doctrine_rest_helper');
         $factory = $this->get('sulu_core.doctrine_list_builder_factory');
 
+        $fieldDescriptors = $this->getFieldDescriptors();
         $listBuilder = $factory->create(self::ENTITY_NAME);
-        $restHelper->initializeListBuilder($listBuilder, $this->getFieldDescriptors());
-        $results = $listBuilder->execute();
+        $restHelper->initializeListBuilder($listBuilder, $fieldDescriptors);
+
+        $ids = null;
+        if (null !== $request->get('ids')) {
+            $ids = array_filter(explode(',', $request->get('ids', '')));
+            $listBuilder->in($fieldDescriptors['id'], $ids);
```

Comment:
In the case where $ids is empty because we deselect in the content page the choice(s), it's necessary to test this before passing $ids to the list
``` if (empty($ids)) { $ids = [0]; }```